core/src/test/scala/akka/persistence/cassandra/EventsByTagRestartSpec.scala

```diff
@@ -83,5 +84,43 @@ class EventsByTagRestartSpec extends CassandraSpec(EventsByTagRestartSpec.config
       tagProbe.expectNoMessage(waitTime)
       tagProbe.cancel()
     }
+
+    "use correct tag sequence nrs when actor is stopped prematurely" in {
+      val queryJournal = PersistenceQuery(system).readJournalFor[CassandraReadJournal](CassandraReadJournal.Identifier)
+      val p2 = system.actorOf(TestTaggingActor.props("p2", Set("green")))
+      val probe1 = TestProbe()
+      probe1.watch(p2)
+      p2.tell("e1", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e2", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e3", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e4", probe1.ref)
+      // the PoisonPill will cause the actor to be stopped before the journal write has been completed,
+      // and is therefore similar to if the CircuitBreaker would trigger
+      p2 ! PoisonPill
+      probe1.expectTerminated(p2)
+
+      Thread.sleep(2000)
```

Comment:
if we just immediately restart I think the bug didn't always manifest

Reply:
I think I'd prefer to have this test without the additional start, and read the 1 to 4 tags here right away.
Then add another test that is also doing the restart but maybe without the sleep.

Reply:
The 4th tag wouldn't be written without the additional start, but ok I'll make it two cases.